### ISSUE TITLE
Provide a way to create a client with an Endpoint rather than with a URI

### ISSUE THREAD

**Issue by trustin** [line/armeria#1231]
A user sometimes would want to use `Endpoint` to create a client especially when he or she wants to specify the IP address of the remote address explicitly:

```java
HelloService.Iface client = new ClientBuilder(
    "tbinary+http", Endpoint.of("foo").withIpAddr(127.0.0.1), "/thrift/hello").build(...);

HttpClient client = HttpClient.of("http", Endpoint.of(...).withIpAddr(...), "/...");
```

This way, a user doesn't have to write the boilerplate code like this:

```java
String ipAddr = endpoint.ipAddr();
if (ipAddr != null) {
    authority = endpoint.authority();
} else if (NetUtil.isIpv4Address(ipAddr)) {
    authority = ipAddr + ':' + port;
} else {
    authority = '[' + ipAddr + "]:" + port;
}
String uri = "http://" + authority + path;
```

**Comment by SeungukHan** [comment]
I am working on this issue.

**Comment by minwoox** [comment]
Closed by #1743
